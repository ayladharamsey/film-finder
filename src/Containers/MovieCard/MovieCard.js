import React, { Component } from 'react';
import { connect } from 'react-redux';
import './MovieCard.scss';
import { faveMovie } from '../../actions/index'
import favorite from '../../images/favorite.svg'
import unfavorite from '../../images/unfavorite.svg'
import { Redirect } from 'react-router-dom'; 
import { setFavorites, getFavorites } from '../../apiCalls'


export class MovieCard extends Component {
  constructor(){
    super()
    this.state = {
      btnClick: false,
    }
  }

  render() {
    const favoriteMovie = (event) => {
      this.props.faveMovie(parseInt(event.target.parentNode.parentNode.id));
      getFavorites(this.props.user.id)
      setFavorites(this.props.user.id, {
        "movie_id": `${event.target.parentNode.parentNode.id}`,
        "title": "sample",
        "poster_path": "example",
        "release_date": "2019-10-18",
        "vote_average": 5.8,
        "overview": "once upon a time"
      } );
    }
  
    const checkLogginStatus = () => {
      if (user.id === undefined) {
        return (
          <Redirect to='/login'>Not Logged In</Redirect>
        )
      }
      return
    }

    const handleClick = (event, id) => {
      this.setState({btnClick: true})
      favoriteMovie(event, id) 
    }

    const { 
    user,
    id, 
    movieRating, 
    backgroundImage, 
    overview, 
    posterImage, 
    title, 
    releaseDate, 
    isFavorited 
    } = this.props;

    const cutOverview = overview.split('').splice(0, 140)

    const { btnClick } = this.state

    let error;

    if ((btnClick === true)) {
      error = checkLogginStatus()
    }
    
    return (
    <article id={id} className="movie-card">
      <img
        className='movieCard__img'
        src={posterImage}
        alt='Official movie poster'
      />
      <div className="footer">
        <p className="over-view">Overview: {cutOverview}</p>
        <p>Release Date: {releaseDate}</p>
        <p className="movie_title">{title}</p>
        {isFavorited ? 
          <img className="fave_btn-img" src={favorite} alt="favorited" onClick={(event) => handleClick(event)}/> : 
          <img className="fave_btn-img" src={unfavorite} alt="un favorited" onClick={(event) => handleClick(event, id)}/>}
        {error}
      </div>
    </article>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  faveMovie: (id) => dispatch (faveMovie(id))
})

export const mapStateToProps = state => ({
  user: state.user
})

export default connect(mapStateToProps, mapDispatchToProps)(MovieCard);

