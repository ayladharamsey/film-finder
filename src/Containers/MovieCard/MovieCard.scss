$style-border: 1px solid red; 

.movie-card {
  margin-left: 50px;
  // border: $style-border; 
  height: 50%; 
  width: 150px; 
  background-color: beige; 
  padding: 15px; 
  box-shadow: 0px 5px 5px 0px #e8e8e8, 0px 10px 10px 0px #e8e8e8;
  border-radius: 7px;

}



