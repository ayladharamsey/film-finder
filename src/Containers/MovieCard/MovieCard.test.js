import React from 'react';
import { shallow } from 'enzyme';
import { MovieCard } from './MovieCard';

describe('MovieCard container tests', () => {

  it('MovieCard should match its snapshot', () => {
    const wrapper = shallow( <MovieCard /> )
    expect(wrapper).toMatchSnapshot();
  })
})