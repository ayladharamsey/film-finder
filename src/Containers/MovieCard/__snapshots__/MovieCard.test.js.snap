// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard container tests MovieCard should match its snapshot 1`] = `ShallowWrapper {}`;
